Sources/NIOExtras/WritePCAPHandler.swift

```diff
@@ -364,31 +365,44 @@ extension NIOWritePCAPHandler: ChannelDuplexHandler {
         defer {
             context.write(data, promise: promise)
         }
-        guard self.closeState == .notClosing else {
-            return
-        }
-        
+
         let data = self.unwrapInboundIn(data)
         self.buffer.clear()
         do {
             var data = data
+            var futureWrittenOutboundBytes = self.writtenOutboundBytes
             while var payloadToSend = self.takeSensiblySizedPayload(buffer: &data) {
                 try self.buffer.writePCAPRecord(.init(payloadLength: payloadToSend.readableBytes,
                                                       src: self.localAddress(context: context),
                                                       dst: self.remoteAddress(context: context),
                                                       tcp: TCPHeader(flags: [],
                                                                      ackNumber: nil,
-                                                                     sequenceNumber: self.writtenOutboundBytes,
+                                                                     sequenceNumber: futureWrittenOutboundBytes,
                                                                      srcPort: .init(self.localAddress(context: context).port!),
                                                                      dstPort: .init(self.remoteAddress(context: context).port!))))
-                self.writtenOutboundBytes += payloadToSend.readableBytes
+                futureWrittenOutboundBytes += payloadToSend.readableBytes
                 self.buffer.writeBuffer(&payloadToSend)
             }
-            self.writeBuffer(self.buffer)
+            self.pendingWrites.append(self.buffer)
```

Comment:
probably, we want to buffer the `PCAPRecord` and not the raw bytes because we should set the timestamp in `flush` and not here.